types/wrapper.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Wrapper is a generic wrapper, with a type field for distinguishing its
+// contents.
+type Wrapper struct {
+	// Type is the fully-qualified type name, e.g.
+	// github.com/sensu/sensu-go/types.Check,
+	// OR, a short-hand name that assumes a package path of
+	// github.com/sensu/sensu-go/types.
+	Type string `json:"type"`
+
+	// Value is a valid Resource of concrete type Type.
+	Value Resource `json:"spec"`
+}
```

Comment:
I believe we should also open an issue so sensuctl uses that `Wrapper` struct when printing a resource in JSON, so we can easily do something like this:

```bash
$ sensuctl check info check-cpu --format json > check-cpu.json
# modify the command or something else
$ sensuctl create -f check-cpu.json
```

Makes any sense?

Reply:
Makes sense to me. Are you thinking that the API would wrap all of its responses?

Reply:
Oh yes, that could be pretty cool! I'll open an issue for that

Reply:
Done: https://github.com/sensu/sensu-go/issues/1221